### ISSUE TITLE
AKS region changes in Edit Mode 

### ISSUE THREAD

**Issue by IsaSih** [rancher/dashboard#11841]
**Setup**
- Rancher version: v2.9-0f2bd9c3cde85aa2b5270946d0cecd5c95b86fd2-head

**Describe the bug**

When creating an AKS cluster in a region different from the default displayed in the UI (East US) with the wrong configuration, once the user edits the cluster config, the Azure Region is set back to the default, undoing the region that the user previously selected

**To Reproduce**
1- Via UI, create an Azure cluster under a region other than East US with incomplete or wrong config
2- After the cluster is created, try to edit the cluster config via UI
3 - Observe that the region switches to the East US

**Result**
the Azure Region is set back to the default, undoing the region that the user previously selected

**Expected Result**
The region previously selected by the user must be preserved in the Edit form

**Screenshots**

https://github.com/user-attachments/assets/65213a34-eaab-4687-84b0-2c6e8b7471ed

**Comment by IsaSih** [comment]
I'll reopen the issue as the behavior is still not expected. Once the user edits the cluster region when it's still in provisioning mode, the dropdown is available for changes, then the user can choose a different region and Save the changes. However, if the user edits the cluster a second time, the original region selected is locked in the dropdown, with a `No options` element being displayed and scrolled in the whole page as observed in the screenshot.

Tested in 
Rancher | v2.11-a1bda936c809e0394520c9d40d2998529223401e-head
Dashboard | master c987984
Rancher CLI | v2.11.0-rc.1

**Comment by IsaSih** [comment]
https://github.com/user-attachments/assets/5c1afd0a-009e-424f-83b9-362dd32ebc0f

**Comment by mantis-toboggan-md** [comment]
That video looks like expected behavior to me. Region is generally only editable immediately after creating an AKS cluster in Rancher, unless something has gone wrong. The intent is to allow users to fix mistakes if some aspect of their cluster configuration has prevented backing resources in Azure to be created, for whatever reason. 

Pretty soon after you create an AKS cluster through Rancher, Rancher will go talk to azure and create AKS resources on your behalf, the state of which are reported back to Rancher (and its UI) in the cluster `status` field. In the period of time between you initially creating your cluster in the Rancher UI, and when stuff starts being created in azure, you can edit most of the cluster spec. This is the stage at which you are changing the region in your video. Once the cluster `status` field reports that there are underlying resources created in azure, less of the form is editable. That's the state the cluster is in at the end of your video.

**Comment by IsaSih** [comment]
I understand your point, but the way the UI behaves gives the impression that changing the region is successful because the field is visible, editable, and the user can save it with no error messages. But then, the field is locked with the initial selection, not the supposedly successful change. So, I believe that we should lock this field unless an error in the provisioning process occurs. Otherwise, it feels like the field can be edited when in fact it cannot.
This is the behavior in 
Rancher | v2.11-967453d983d9927ec0880b549f798d99ad354cbf-head
Dashboard | master 16e73b1
Rancher CLI | v2.11.0-rc.1

[Rancher](https://github.com/rancher/rancher)	v2.11-967453d983d9927ec0880b549f798d99ad354cbf-head
[Dashboard](https://github.com/rancher/dashboard)	master 16e73b1
[Rancher CLI](https://github.com/rancher/cli)	v2.11.0-rc.1
https://github.com/user-attachments/assets/ed9e5951-abf8-4279-9ead-6b9d8ac8a0c4

**Comment by mantis-toboggan-md** [comment]
>  So, I believe that we should lock this field unless an error in the provisioning process occurs. Otherwise, it feels like the field can be edited when in fact it cannot. 


I agree that this would be preferable, but it is not so simple to implement. The big problem is that clusters that stall out during provisioning don't always report errors back to Rancher nicely: they may linger in a "Provisioning" state indefinitely. Being able to edit clusters stuck in a 'provisioning' state is a pretty helpful tool: it's not worthwhile to get rid of this functionality just to avoid the behavior you're seeing here which, again, while not ideal, is only possible if editing a cluster _right_ before the stage of provisioning where resources are created in aks -- in my experience that window is a couple seconds.

**Comment by IsaSih** [comment]
Thanks for the clarification Nancy! I agree that this involves a complex solution that involves limiting a more valuable functionality. So, I'm closing this issue as resolved.
